lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -7173,12 +7174,15 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
     case AdjointValueKind::Concrete: {
       auto val = av.getConcreteValue();
       unsigned adjIdx = 0;
-      auto elts = builder.createDestructureTuple(ti->getLoc(), val);
+      auto valCopy = builder.emitCopyValueOperation(ti->getLoc(), val);
+      auto elts = builder.createDestructureTuple(ti->getLoc(), valCopy);
+      for (auto elt : elts->getResults())
+        recordTemporary(elt);
       for (auto i : range(ti->getNumOperands())) {
         if (!getTangentSpace(ti->getOperand(i)->getType().getASTType()))
           continue;
-        auto adjElt = val;
-        if (val->getType().is<TupleType>())
+        auto adjElt = valCopy;
```

Comment:
Note: these three lines of code can be simplified, https://github.com/apple/swift/pull/28253#discussion_r346158296:
```suggestion
        auto adjElt = elts->getResult(adjIdx++);
```

https://github.com/apple/swift/pull/28253 should be merged soon so there'll be a small conflict.